src/ucm/bistro/bistro.c

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.       ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <dlfcn.h>
+#include <stdlib.h>
+
+#include <ucm/bistro/bistro.h>
+#include <ucm/bistro/bistro_int.h>
+
+#if HAVE___CLEAR_CACHE
+void __clear_cache(char* beg, char* end);
+#endif
+
+ucs_status_t ucm_bistro_remove_restore_point(ucm_bistro_restore_point_h rp)
```

Comment:
why need this function? just call free()...

Reply:
malloc/free pairs of calls, in case if we add more functionality it may require to free more resources than just memory pool (originally I was going to add restoring of all patched calls on finalise, but it makes PR too complicated)